pkgs/applications/networking/cisco-packet-tracer/8.nix

```diff
@@ -21,106 +22,96 @@
 , makeWrapper
 , nspr
 , nss
+, qt5
 , requireFile
 , xorg
 }:
 
 let
-  version = "8.2.0";
-
-  ptFiles = stdenv.mkDerivation {
-    name = "PacketTracer8Drv";
-    inherit version;
-
-    dontUnpack = true;
-    src = requireFile {
-      name = "CiscoPacketTracer_${builtins.replaceStrings ["."] [""] version}_Ubuntu_64bit.deb";
-      sha256 = "1b19885d59f6130ee55414fb02e211a1773460689db38bfd1ac7f0d45117ed16";
-      url = "https://www.netacad.com";
-    };
-
-    nativeBuildInputs = [
-      alsa-lib
-      autoPatchelfHook
-      dbus
-      dpkg
-      expat
-      fontconfig
-      glib
-      libdrm
-      libglvnd
-      libpulseaudio
-      libudev0-shim
-      libxkbcommon
-      libxml2
-      libxslt
-      makeWrapper
-      nspr
-      nss
-    ] ++ (with xorg; [
-      libICE
-      libSM
-      libX11
-      libxcb
-      libXcomposite
-      libXcursor
-      libXdamage
-      libXext
-      libXfixes
-      libXi
-      libXrandr
-      libXrender
-      libXScrnSaver
-      libXtst
-      xcbutilimage
-      xcbutilkeysyms
-      xcbutilrenderutil
-      xcbutilwm
-    ]);
-
-    installPhase = ''
-      dpkg-deb -x $src $out
-      chmod 755 "$out"
-      makeWrapper "$out/opt/pt/bin/PacketTracer" "$out/bin/packettracer" \
-        --prefix LD_LIBRARY_PATH : "$out/opt/pt/bin"
-
-      # Keep source archive cached, to avoid re-downloading
-      ln -s $src $out/usr/share/
-    '';
+  hashes = {
+    "8.2.0" = "1b19885d59f6130ee55414fb02e211a1773460689db38bfd1ac7f0d45117ed16";
+    "8.2.1" = "1fh79r4fnh9gjxjh39gcp4j7npgs5hh3qhrhx74x8x546an3i0s2";
```

Comment:
Any specific reason why to have both hashes present?

Reply:
If some class require a specific packet tracer version so you can easily use withVersion to use that

Reply:
It is preferable to do an `overrideAttrs` on src and version if someone really wants the old ones.

Reply:
Yes, but this way is simpler to use a older version

Note that only the latest version is evaluated by default.

Reply:
What do you plan to do if version `8.2.2` or later is released?

Reply:
Change mkDerivation to 8.2.2 then add 8.2.2 hash there

r-ryantm can't update this package automatically so someone (probably me) will have to do this manually anyway

Reply:
While I don't really agree with this, we have Coq that literally maintains all versions from 8.5 to 8.17, so I can't say that this pattern doesn't fit in nixpkgs, it is just very, very rare...

Reply:
Hmmmm I see! That makes sense, thanks for clarifying

Reply:
> Yes, but this way is simpler to use a older version

That is not the goal of nixpkgs. This hash is unused and dead code which is to be removed.

Reply:
That's for withVersion

Reply:
We don't want to collect hashes here for versions which might be used by some override not in tree. If the application is so picky about versions and so bad at maintaining backwards compatibility then we could add more versions of it but I am not sure if we need this. We already have two major versions, isn't that enough?